pkg/network/ebpf/c/prebuilt/http.c

```diff
@@ -622,44 +626,69 @@ char _license[] SEC("license") = "GPL"; // NOLINT(bugprone-reserved-identifier)
 
 // GO TLS PROBES
 
+static __always_inline tls_offsets_data_t* get_offsets_data() {
+    struct task_struct *t = (struct task_struct *) bpf_get_current_task();
+    struct mm_struct *mm;
+    struct file *exe_file;
+    struct dentry *dentry;
+
+    bpf_probe_read(&mm, sizeof(mm), &t->mm);
+    bpf_probe_read(&exe_file, sizeof(exe_file), &mm->exe_file);
+    bpf_probe_read(&dentry, sizeof(dentry), &exe_file->f_path.dentry);
```

Comment:
It will compile, but of course may or may not behave correctly if offsets for those fields are different in another kernel version. This will work only for runtime compiled code or with CO-RE.

One possibility: make this feature only available if runtime compilation is enabled.

Reply:
For now it will be runtime compiled only at first. A prebuilt version will come in the future, if possible (i.e depends on the feasibility of adding the necessary offset guessing additions).